src/string/strlen.rs

```diff
@@ -1,14 +1,46 @@
-use core::usize;
+use core::{ usize, mem };
 
 use c_types::*;
 
 #[no_mangle]
-pub unsafe extern "C" fn strlen(s: *const c_schar) -> size_t {
-    // TODO(adam) convert to checking word-size chunks
-    for i in 0.. {
-        if *s.offset(i) == 0 {
-            return i as usize;
+pub unsafe extern "C" fn strlen(mut s: *const c_schar) -> size_t {
+    
+    #[inline]
+    fn align() -> usize { mem::size_of::<size_t>() }
+    
+    #[inline]
+    fn ones() -> usize { !0usize }
+    
+    #[inline]
+    fn highs() -> usize { usize::wrapping_mul(ones(), usize::MAX)/2 + 1 }
+    
+    #[inline]
+    fn has_zero(x: usize) -> bool {
+        usize::wrapping_sub(x, ones() & !x & highs()) != 0
```

Comment:
Ah, but you're looking for zero *bytes*. Maybe call the function `has_zero_byte`? ;-)